client/my-sites/invites/invite-accept/index.jsx

```diff
@@ -49,32 +49,34 @@ class InviteAccept extends React.Component {
 		matchEmailError: false,
 	};
 
-	UNSAFE_componentWillMount() {
+	componentDidMount() {
 		// The site ID and invite key are required, so only fetch if set
 		if ( this.props.siteId && this.props.inviteKey ) {
-			fetchInvite( this.props.siteId, this.props.inviteKey );
+			this.fetchInvite();
```

Comment:
A couple of questions here:

* Should we abort the request if the user navigates away?
* If this component is mounted twice in a relatively short period of time, will the request be doubled? This is related to my comment about migrating to the data layer, as that'd solve such a potential problem.

Reply:
@tyxla The `invites-accept-validation` store is used only in this component, and stores only the response from one `fetchInvite` request. The invite object ceases to exist once it's successfully accepted, so it doesn't need to be cached and stored by `siteId` and `inviteKey`.

That's probably why @flootr decided to make it a local state of the component. Data layer would require a Redux store for the state.

> Should we abort the request if the user navigates away?

That's valid -- there should be an `.isMounted` check before calling `.setState` in the promise callbacks.

> If this component is mounted twice in a relatively short period of time, will the request be doubled?

That shouldn't be a concern for a request that only reads stuff and doesn't do modifications.

The data layer deduplication is useful mainly if there are two independent components on a page that query or use the same data. Then it makes sure the requests are co-ordinated and the responses are shared.

Reply:
Thanks for chiming in @jsnajdr. Would d8bebe1 work in that regard?